### PR TITLE
Simplify handling providers

### PR DESCRIPTION
**Pull request by invidian**
As a prerequisite for simplified version of #137.

### CODE REVIEW

File: aws/flatcar-linux/kubernetes/require.tf [hunk 1]
```diff
@@ -2,24 +2,12 @@
...
-}
-
-provider "template" {
-  version = "~> 2.1"
-}
 
-provider "tls" {
-  version = "~> 2.0"
+  required_providers {
+    aws      = ">= 2.31.0"
```
Comment (rata) [inline aws/flatcar-linux/kubernetes/require.tf]: Doesn't this allow version 3.0.0? I don't think we want that, as that would mean the aws provider probably broke compatibility. Why not maintain the symbols used before?

Comment (invidian) [reply aws/flatcar-linux/kubernetes/require.tf]: Oops, bad copy-paste. will fix.

Comment (invidian) [reply aws/flatcar-linux/kubernetes/require.tf]: Fixed.

File: aws/flatcar-linux/kubernetes/require.tf [hunk 2]
```diff
@@ -2,24 +2,12 @@
...
-}
 
-provider "tls" {
-  version = "~> 2.0"
+  required_providers {
+    aws      = "2.31.0"
+    local    = "~> 1.2"
+    null     = "~> 2.1"
+    template = "~> 2.1"
+    tls      = "~> 2.0"
```
Comment (rata) [inline aws/flatcar-linux/kubernetes/require.tf:12]: no ct provider needs to be added here? (I see you added it on bare-metal)

File: packet/flatcar-linux/kubernetes/require.tf [hunk 3]
```diff
@@ -2,32 +2,14 @@
...
-}
-
-provider "packet" {
-  version = "~> 2.7.3"
-}
 
-provider "aws" {
-  version = "2.31.0"
+  required_providers {
+    aws      = "2.31.0"
```
Comment (rata) [inline packet/flatcar-linux/kubernetes/require.tf]: I the think previous (deleted) missing `~>` was probably a bug introduced recently (tf 0.12 migration?)

File: packet/flatcar-linux/kubernetes/workers/require.tf [hunk 4]
```diff
@@ -2,24 +2,14 @@
...
-}
-
-provider "tls" {
-  version = "~> 2.0"
-}
 
-provider "packet" {
-  version = "~> 2.7.3"
+  required_providers {
+    aws      = "2.31.0"
```
Comment (rata) [inline packet/flatcar-linux/kubernetes/workers/require.tf]: idem

**Review by rata — APPROVED** [review]
LGTM.

There are some small things that can be fixed while we are there , I think, but is reasonable in any case. If you prefer not to fix the comments and makes sense to address them for you too, please create an issue for them :)

File: aws/flatcar-linux/kubernetes/require.tf (hunk 2 above)
Comment (invidian) [reply aws/flatcar-linux/kubernetes/require.tf:12]: It wasn't there, but yeah, it should be added. Adding in a separate commit.

File: packet/flatcar-linux/kubernetes/require.tf (hunk 3 above)
Comment (invidian) [reply packet/flatcar-linux/kubernetes/require.tf]: Could be, but I'd say it's out of scope for this PR to fix that?

Comment (invidian) [reply packet/flatcar-linux/kubernetes/require.tf]: Created #143 for it.

File: packet/flatcar-linux/kubernetes/workers/require.tf (hunk 4 above)
Comment (invidian) [reply packet/flatcar-linux/kubernetes/workers/require.tf]: Created #143 for it.

File: packet/flatcar-linux/kubernetes/require.tf (hunk 3 above)
Comment (rata) [reply packet/flatcar-linux/kubernetes/require.tf]: :ok_hand:

**Comment by invidian** [comment]
Updated:
- I split changes per platform, so they are easier for review.
- Checked all platforms for missing constraints and added them where needed.
- Make sure docs are updated as well.

File: packet/flatcar-linux/kubernetes/require.tf (hunk 3 above)
Comment (mauriciovasquezbernal) [reply packet/flatcar-linux/kubernetes/require.tf]: > I the think previous (deleted) missing `~>` was probably a bug introduced recently (tf 0.12 migration?)

It wasn't me :stuck_out_tongue_closed_eyes:

**Review by surajssd — APPROVED** [review]
LGTM, this has reduced a lot of unnecessary boilerplate \o/

**Review by pothos — APPROVED** [review]

**Comment by invidian** [comment]
Will rebase and merge after #146.

**Pull request merged**